Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -13,35 +13,62 @@
 
 use Neos\Cqrs\Event\EventInterface;
 use Neos\Cqrs\Event\EventTransport;
-use Neos\Cqrs\Event\EventType;
+use Neos\Cqrs\Event\EventTypeService;
 use Neos\Cqrs\Message\MessageMetadata;
 use TYPO3\Flow\Annotations as Flow;
-use TYPO3\Flow\Utility\Arrays;
 
 /**
  * Base class for an aggregate root
  */
 abstract class AbstractAggregateRoot implements AggregateRootInterface
 {
     /**
+     * @var EventTypeService
+     * @Flow\Inject
+     */
+    protected $eventTypeService;
+
+    /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateIdentifier;
 
     /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateName;
 
     /**
      * @var EventTransport[]
-     * @Flow\Transient
      */
     protected $events = [];
 
     /**
+     * When calling self::recordThatMethod in the contructor
+     * the object is not initialized so event are queued
+     *
+     * @var array
+     */
+    protected $pendingEvents = [];
+
+    /**
+     * @var bool
+     */
+    protected $initialized = false;
+
+    /**
+     * Register event listeners based on annotations
+     */
+    public function initializeObject()
+    {
+        $this->initialized = true;
+        foreach ($this->pendingEvents as $data) {
```

Comment:
This is need because recordThat can be called inside the aggregate constructor. If the apply method depend on property injection, the apply method will fail. So we simple put event in a queue, and record them again after the object full initialization. I know it's not common to have dependency in aggregate, but I think it's import to support the property injection without restriction. 

That was needed for the PR to support the injection of the EventTypeService